src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -20,18 +20,22 @@ function toWeb3Connection<T extends Connector>([connector, hooks]: [T, Web3React
   return [connector, hooks]
 }
 
-function getWallet(provider?: JsonRpcProvider | Eip1193Provider) {
+function getWallet(setError: (error: Error) => void, provider?: JsonRpcProvider | Eip1193Provider) {
```

Comment:
Specify the error handler last, and add a return type, so the signature should be

    function getWalletFromProvider(provider: JsonRpcProvider | Eip1193Provider | undefined, setError: (error: Error) => void): Web3Connection {

This puts the main argument, `provider`, in a more prominent/obvious position.